File: 조서영/2주차/15651.py
# 15651, Silver 3
# https://www.acmicpc.net/problem/15651

import sys

N,M=map(int,sys.stdin.readline().split())

S=[0 for i in range(M)]
def dfs(idx):
    if idx==M:
        print(' '.join(S))
        return  
    
    for i in range(1, N+1, 1):
        S[idx]=str(i)
        dfs(idx+1)

dfs(0)

File: 장익환/2주차/15651.cpp
/*
백준 15651번, https://www.acmicpc.net/problem/15651
*/
#include <iostream>
using namespace std;

int N, M;
int seq[7];   // 수열을 저장할 배열

void find_sequence(int k) {
	if (k == M) {             // 레벨이 M이 되면 
		// 수열을 찾았으므로 출력
		for (int i = 0; i < M; i++) {
			cout << seq[i] << ' ';
		}
		cout << "\n";
		return; 
	}

	// 중복해서 선택할 수 있으므로 1 ~ N 중에서 선택
	for (int i = 1; i <= N; i++) {
		seq[k] = i;          
		find_sequence(k + 1);
	}
}

int main(void) {
	cin >> N >> M;
	
	find_sequence(0);
}
